language/move-vm/types/src/values/values_impl.rs

```diff
@@ -1884,6 +1884,188 @@ impl Display for Locals {
     }
 }
 
+pub mod debug {
```

Comment:
this is all good but I would really think about how to make all of this just the printing/display utilities for move values.
In other words the debug module should just be a call into a "display" function.
I think we want that and the stack/environment (core dump) printing as a core part of our display/debug story and it would be nice to see all of that work together.
Those are considerations and no particular action to take here

Reply:
So here we have two kinds of print/Display functions: debug::print with Type (the one you are seeing right now) and Display without Type. My understanding is that debug::print is facing Move developers and should be optimized for user experience. The rust Debug/Display dumps are more for ourselves to inspect a VM crash or whatever. Now what's unclear to me is that once we have debug::print, do we still need Debug/Display.

Reply:
I'd say we want a unified story for all kind of printing and let client decide what they want to use. And by client I mean internally us. That is, we call the most sensible API for whatever it is that we are doing.
Again though I don't think we should do it in this diff but it seems a good thing to have to me.
Thoughts?